src/goto-instrument/cover.cpp

```diff
@@ -88,11 +120,117 @@ class basic_blockst
     block_line_cover_mapt;
   block_line_cover_mapt block_line_cover_map;
 
-  inline unsigned operator[](goto_programt::const_targett t)
+  /// \param t a goto instruction
+  /// \return the block number of the block
+  ///         the given goto instruction is part of
+  unsigned block_of(goto_programt::const_targett t)
   {
     return block_map[t];
   }
 
+  /// \param block_nr a block number
+  /// \return the instruction that has been selected for
+  ///   instrumentation representative of the given block
+  goto_programt::const_targett instruction_of(
+    unsigned block_nr)
+  {
+    return block_loc[block_nr];
+  }
+
+  /// Select an instruction to be instrumented for each basic block such that
+  /// the java bytecode indices for each basic block is unique
+  /// \param goto_program The goto program
+  void select_unique_java_bytecode_indices(const goto_programt &goto_program)
+  {
+    std::set<unsigned> blocks_seen;
+    std::set<irep_idt> bytecode_indices_seen;
+
+    forall_goto_program_instructions(it, goto_program)
+    {
+      unsigned block_nr=block_map[it];
+      if(blocks_seen.find(block_nr)!=blocks_seen.end())
```

Comment:
For `std::map` it will return quickly since it knows the count is 0 or 1. I'd always seen `count(...)` as the standard membership test for STL maps and sets, purely for being a bit briefer than find and avoiding needlessly making-then-comparing an iterator, but shrug, don't mind that much either way